lua/telescope/actions/init.lua

```diff
@@ -73,71 +73,66 @@ function actions._goto_file_selection(prompt_bufnr, command)
+    filename = entry.path or entry.filename
+
+    -- TODO: Check for off-by-one
+    row = entry.row or entry.lnum
+    col = entry.col
+  elseif not entry.bufnr then
+    -- TODO: Might want to remove this and force people
+    -- to put stuff into `filename`
+    local value = entry.value
+    if not value then
+      print("Could not do anything with blank line...")
+      return
     end
 
-    local preview_win = state.get_status(prompt_bufnr).preview_win
-    if preview_win then
-      a.nvim_win_set_config(preview_win, {style = ''})
+    if type(value) == "table" then
+      value = entry.display
     end
 
-    local entry_bufnr = entry.bufnr
+    local sections = vim.split(value, ":")
 
-    actions.close(prompt_bufnr)
+    filename = sections[1]
+    row = tonumber(sections[2])
+    col = tonumber(sections[3])
+  end
 
-    if entry_bufnr then
-      if command == 'edit' then
-        vim.cmd(string.format(":buffer %d", entry_bufnr))
-      elseif command == 'new' then
-        vim.cmd(string.format(":sbuffer %d", entry_bufnr))
-      elseif command == 'vnew' then
-        vim.cmd(string.format(":vert sbuffer %d", entry_bufnr))
-      elseif command == 'tabedit' then
-        vim.cmd(string.format(":tab sb %d", entry_bufnr))
-      end
-    else
-      filename = path.normalize(vim.fn.fnameescape(filename), vim.fn.getcwd())
-
-      local bufnr = vim.api.nvim_get_current_buf()
-      if filename ~= vim.api.nvim_buf_get_name(bufnr) then
-        vim.cmd(string.format(":%s %s", command, filename))
-        bufnr = vim.api.nvim_get_current_buf()
-        a.nvim_buf_set_option(bufnr, "buflisted", true)
-      end
+  local entry_bufnr = entry.bufnr
+
+  actions.close(prompt_bufnr)
+
+  if entry_bufnr then
+    if command == 'edit' then
+      vim.cmd(string.format(":buffer %d", entry_bufnr))
+    elseif command == 'new' then
+      vim.cmd(string.format(":sbuffer %d", entry_bufnr))
+    elseif command == 'vnew' then
+      vim.cmd(string.format(":vert sbuffer %d", entry_bufnr))
+    elseif command == 'tabedit' then
+      vim.cmd(string.format(":tab sb %d", entry_bufnr))
+    end
+  else
+    filename = path.normalize(vim.fn.fnameescape(filename), vim.fn.getcwd())
+    local filename_bufnr = vim.fn.bufnr(filename)
+
+    -- check if we didn't pick a different buffer
+    -- prevents restarting lsp server
+    if state.get_global_key('original_buf_id') ~= filename_bufnr then
+      vim.cmd(string.format("%s %s", command, filename))
+    end
 
-      if row and col then
-        local ok, err_msg = pcall(a.nvim_win_set_cursor, 0, {row, col})
-        if not ok then
-          log.debug("Failed to move to cursor:", err_msg, row, col)
-        end
+    if row and col then
+      local ok, err_msg = pcall(a.nvim_win_set_cursor, 0, {row, col})
+      if not ok then
+        log.debug("Failed to move to cursor:", err_msg, row, col)
       end
     end
-    vim.api.nvim_command("doautocmd filetypedetect BufRead " .. vim.fn.fnameescape(filename))
```

Comment:
Not having that breaks some things in the past. See here. I also said that this doesn't happen anymore so yeah. Maybe its fine to remove
https://github.com/nvim-telescope/telescope.nvim/issues/481#issuecomment-771453930

Reply:
That doesn't seem to happen anymore. Also as I said above this pr will make it so that if the buffer is already loaded, it will not reload it again so the buffer local variables should not be reset. It will only load new buffers that haven't already been loaded.

Reply:
I will test this now. But i think its also no longer necessary.